<launch>
  <arg name="mav_name" default="tarot"/> <!--iris-->
  <arg name="fcu_url" default="udp://:14540@127.0.0.1:14557"/>
  <arg name="gcs_url" default="" />
  <arg name="tgt_system" default="1" />
  <arg name="tgt_component" default="1" />
  <arg name="command_input" default="2" />
  <arg name="gazebo_simulation" default="true" />
  <arg name="visualization" default="true"/>
  <arg name="log_output" default="screen" />
  <arg name="fcu_protocol" default="v2.0" />
  <arg name="respawn_mavros" default="false" />
  <arg name="mav_v" default="5"/>
  <arg name="mav_a" default="3.0"/>
  <arg name="mav_ang_v" default="1.5"/>
  <arg name="mav_ang_a" default="0.1"/>
  <arg name="current_ref_pose_as_start" default="false"/>
  <arg name="dt" default="0.01"/>
  <arg name="map" default="$(find mavlink_sitl_gazebo)/worlds/empty.world"/>
  
  <node pkg="geometric_controller" type="geometric_controller_node" name="geometric_controller" output="screen">
          <param name="mav_name" type="string" value="$(arg mav_name)" />
          <remap from="command/bodyrate_command" to="/mavros/setpoint_raw/attitude"/>
          <remap from="/planning/pos_cmd" to="/drone_0_planning/pos_cmd"/>
          <param name="ctrl_mode" value="2.0" />
          <param name="enable_sim" value="$(arg gazebo_simulation)" />
          <param name="enable_gazebo_state" value="true"/>
          <param name="max_acc" value="4.0" />
          <!--Params specific for Iris copter F = 0.0516 -->
          <param name="attctrl_constant" value="0.3"/>
          <param name="normalizedthrust_constant" value="0.03"/>
          <param name="normalizedthrust_offset" value="0.0"/>
          <param name="landing_vel" value="0.35" />
          <param name="Kp_x" value="2.0"/>
          <param name="Kp_y" value="2.0"/>
          <param name="Kp_z" value="3.0"/>
          <param name="Kv_x" value="6.0"/>
          <param name="Kv_y" value="6.0"/>
          <param name="Kv_z" value="8.0"/>
          <param name="Krp" value="3.0"/>
          <param name="Kyaw" value="4.5"/>
          <param name="init_pos_x" value="-0.0"/>
          <param name="init_pos_y" value="-0.0"/>
          <param name="init_pos_z" value="1.0"/>
  </node>

  <!-- Launch rqt_reconfigure -->
  <node pkg="rqt_reconfigure" type="rqt_reconfigure" output="screen" name="rqt_reconfigure" />

  <include file="$(find mavros)/launch/node.launch">
      <arg name="pluginlists_yaml" value="$(find mavros)/launch/px4_pluginlists.yaml" />
      <arg name="config_yaml" value="$(find mavros)/launch/px4_config.yaml" />
      
      <arg name="fcu_url" value="$(arg fcu_url)" />
      <arg name="gcs_url" value="$(arg gcs_url)" />
      <arg name="tgt_system" value="$(arg tgt_system)" />
      <arg name="tgt_component" value="$(arg tgt_component)" />
      <arg name="log_output" value="$(arg log_output)" />
      <arg name="fcu_protocol" value="$(arg fcu_protocol)" />
      <arg name="respawn_mavros" default="$(arg respawn_mavros)" />
  </include>

  <include file="$(find px4)/launch/posix_sitl.launch">
      <arg name="vehicle" value="$(arg mav_name)"/>
      <arg name="world" default="$(arg map)"/>
      <arg name="x" value="0"/>
      <arg name="y" value="0"/>
      <arg name="z" value="0.5"/>0.02
      <arg name="R" value="0"/>
      <arg name="P" value="0.2"/>
      <!-- p = 0.5 solo ,z =0.5 -->
      <arg name="Y" value="0"/>
  </include>
  <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="0.0 0.0 0 -0.5 0.5 -0.5 0.5 base_link camera_depth_optical_frame 0.01" />
  <node pkg="tf" type="static_transform_publisher" name="link2_broadcaster" args="0 0 0 0 0 0 1 map world 0.01" />
  <node pkg="geometric_controller" type="load" name="camera_odom" output="screen"/>
  <!-- <node pkg="tf" type="static_transform_publisher" name="link2_broadcaster" args="0 0 0.2 0 0 0 1 base_link camera_link 0.01" /> -->
  <!-- <group if="$(arg visualization)">
      <node type="rviz" name="rviz" pkg="rviz" args="-d $(find geometric_controller)/launch/config_file.rviz" />
  </group> -->

</launch>
